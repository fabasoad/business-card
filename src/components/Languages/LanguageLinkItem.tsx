import * as React from 'react'
import LanguageItem from './LanguageItem'

export default function LaguageLinkItem({ code, text, url }) {
  return (
    <LanguageItem code={code}>
      <a className="h5 mt-2" target="_blank" rel="noopener noreferrer" href={url}>{text}</a>
    </LanguageItem>
  )
}