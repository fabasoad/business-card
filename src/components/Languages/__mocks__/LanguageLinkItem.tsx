import * as React from 'react'
import { LanguageLinkItemProps } from '../LanguageLinkItem'

export default function LanguageLinkItem({ code, text, url }: LanguageLinkItemProps) {
  return (
    <div data-testid={code}>
      <div data-testid="text">{text}</div>
      <div data-testid="url">{url}</div>
    </div>
  )
}
