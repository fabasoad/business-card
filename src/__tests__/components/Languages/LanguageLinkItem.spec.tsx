/// <reference types="jest" />
import * as React from 'react'
import { shallow, ShallowWrapper } from 'enzyme'
import LanguageLinkItem from '../../../components/Languages/LanguageLinkItem'

test('should render LanguageLinkItem correctly', () => {
  const wrapper: ShallowWrapper = shallow(<LanguageLinkItem
    code="gb"
    text="test-text"
    url="test-url"
  />)
  expect(wrapper).toMatchSnapshot()
})
