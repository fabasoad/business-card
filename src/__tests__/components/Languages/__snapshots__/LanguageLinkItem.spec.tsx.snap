// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render LanguageLinkItem correctly 1`] = `
<LanguageItem
  code="gb"
>
  <a
    className="h5 mt-2"
    href="test-url"
    rel="noopener noreferrer"
    target="_blank"
  >
    test-text
  </a>
</LanguageItem>
`;
